### PR TITLE
Static cast screen column count for scan alert 559

### PR DESCRIPTION
**Pull request by sithlord48**
Resolve: https://github.com/deskflow/deskflow/security/code-scanning/559

### CODE REVIEW

File: src/gui/src/ScreenSetupModel.cpp [hunk 1]
```diff
@@ -39,7 +39,8 @@ ScreenSetupModel::ScreenSetupModel(ScreenList &screens, int numColumns, int numR
     return;
   }
 
-  if (m_NumColumns * m_NumRows > screens.size()) {
+  long span = static_cast<long>(m_NumColumns) * m_NumRows;
```
Comment (nbolton) [inline src/gui/src/ScreenSetupModel.cpp:42]:
```suggestion
  const auto span = static_cast<long>(m_NumColumns) * m_NumRows;
```

Comment (nbolton) [reply src/gui/src/ScreenSetupModel.cpp:42]: Minor.

**Review by nbolton — APPROVED** [review]

**Comment by nbolton** [comment]
Commit message suggestion:
> fix: Static cast screen column count for scan alert #559

**Comment by sithlord48** [comment]
i was  hoping git would attach the proper issue but tagging code scan you need a full link :(

File: src/gui/src/ScreenSetupModel.cpp (hunk 1 above)
Comment (sithlord48) [reply src/gui/src/ScreenSetupModel.cpp:42]: we don't want auto in this case as the issue is that the larger result can overflow if stored in a temp `int32_t` and not forced into the `long` bounds.

**Review by nbolton — APPROVED** [review]

**Pull request merged**